Start paper UI requires extra mvn install

Following the [Smarthome IDE Setup Guide](http://www.eclipse.org/smarthome/documentation/development/ide.html) explains how to get cleanly compiled Java projects.

However, starting the runtime using the provided Run Configuration will fail starting the Paper UI. Though it registers the paper UI servlet correctly against the path /ui to the ressource folder "web", there is no folder "web". Consequently I get a `HTTP ERROR 404 Problem accessing /ui. Reason: ProxyServlet: /ui `

[This forum post](https://www.eclipse.org/forums/index.php/t/1076298/) suggests to manually run mvn install on the paperui-project in order to generate the web-folder.

However, I don't get it to run: calling Maven from within Eclipse returns the following output

````
[INFO] Scanning for projects...
[INFO] Computing target platform for MavenProject: org.eclipse.smarthome.extension.ui:org.eclipse.smarthome.ui.paper:0.9.0-SNAPSHOT @ C:\eclipse\eclipse-smarthome2\git\smarthome\extensions\ui\org.eclipse.smarthome.ui.paper\pom.xml
[INFO] Fetching p2.index from http://download.eclipse.org/tools/orbit/downloads/drops/R20140525021250/repository/ (0B at 0B/s)
[INFO] Adding repository http://download.eclipse.org/tools/orbit/downloads/drops/R20140525021250/repository
[INFO] Fetching p2.index from http://download.eclipse.org/eclipse/updates/4.4/ (0B at 0B/s)
[INFO] Adding repository http://download.eclipse.org/eclipse/updates/4.4
[INFO] Fetching p2.index from http://download.eclipse.org/modeling/tmf/xtext/updates/releases/2.9.2/ (0B at 0B/s)
[INFO] Adding repository http://download.eclipse.org/modeling/tmf/xtext/updates/releases/2.9.2
[INFO] Adding repository http://download.eclipse.org/modeling/emft/mwe/updates/releases/2.8.3
[INFO] Adding repository http://download.eclipse.org/modeling/emf/emf/updates/2.11/core/R201506010402
[INFO] Adding repository http://dist.springsource.org/snapshot/GRECLIPSE/e4.6
[INFO] Fetching p2.index from http://download.eclipse.org/releases/luna/201406250900/ (0B at 0B/s)
[INFO] Adding repository http://download.eclipse.org/releases/luna/201406250900
[INFO] Adding repository http://www.jupnp.org/update-site/2.2.0
[INFO] Adding repository http://hstaudacher.github.io/osgi-jax-rs-connector/5.3.0
[INFO] Adding repository http://download.eclipse.org/paho
[INFO] Adding repository http://eclipse.github.io/smarthome/third-party/target/repository
[INFO] Adding repository http://www.jmdns.org/update-site
[INFO] Adding repository http://download.eclipse.org/jetty/updates/jetty-bundles-9.x/9.2.9.v20150224
[WARNING] Target location type 'Directory' is not supported
[INFO] Resolving dependencies of MavenProject: org.eclipse.smarthome.extension.ui:org.eclipse.smarthome.ui.paper:0.9.0-SNAPSHOT @ C:\eclipse\eclipse-smarthome2\git\smarthome\extensions\ui\org.eclipse.smarthome.ui.paper\pom.xml
[INFO] Resolving class path of MavenProject: org.eclipse.smarthome.extension.ui:org.eclipse.smarthome.ui.paper:0.9.0-SNAPSHOT @ C:\eclipse\eclipse-smarthome2\git\smarthome\extensions\ui\org.eclipse.smarthome.ui.paper\pom.xml
[INFO]                                                                         
[INFO] ------------------------------------------------------------------------
[INFO] Building Eclipse SmartHome Paper UI 0.9.0-SNAPSHOT
[INFO] ------------------------------------------------------------------------
[INFO] 
[INFO] --- tycho-packaging-plugin:0.23.1:build-qualifier (default-build-qualifier) @ org.eclipse.smarthome.ui.paper ---
[INFO] The project's OSGi version is 0.9.0.201610312126
[INFO] 
[INFO] --- tycho-packaging-plugin:0.23.1:validate-id (default-validate-id) @ org.eclipse.smarthome.ui.paper ---
[INFO] 
[INFO] --- tycho-packaging-plugin:0.23.1:validate-version (default-validate-version) @ org.eclipse.smarthome.ui.paper ---
[INFO] 
[INFO] --- build-helper-maven-plugin:1.8:add-source (add-source) @ org.eclipse.smarthome.ui.paper ---
[INFO] Source directory: C:\eclipse\eclipse-smarthome2\git\smarthome\extensions\ui\org.eclipse.smarthome.ui.paper\src\main\groovy added.
[INFO] 
[INFO] --- frontend-maven-plugin:1.0:install-node-and-npm (Install node and npm) @ org.eclipse.smarthome.ui.paper ---
[INFO] Node v5.4.0 is already installed.
[INFO] NPM 3.3.12 is already installed.
[INFO] 
[INFO] --- frontend-maven-plugin:1.0:npm (npm install) @ org.eclipse.smarthome.ui.paper ---
[INFO] Running 'npm install' in C:\eclipse\eclipse-smarthome2\git\smarthome\extensions\ui\org.eclipse.smarthome.ui.paper
[WARNING] npm WARN addRemoteGit git+https://github.com/jquery/jquery-ui.git#1.11.2 pre-existing cached repo C:\eclipse\eclipse-smarthome2\git\smarthome\extensions\ui\org.eclipse.smarthome.ui.paper\target\npm_cache\_git-remotes\git-https-github-com-jquery-jquery-ui-git-1-11-2-5b6c4032fd076368d4b39dec471a4d07 points to https://github.com/eclipse/smarthome.git and not https://github.com/jquery/jquery-ui.git
[WARNING] npm WARN engine bootstrap@3.3.2: wanted: {"node":"~0.10.1"} (current: {"node":"5.4.0","npm":"3.3.12"})
[ERROR] npm ERR! git clone --template=C:\eclipse\eclipse-smarthome2\git\smarthome\extensions\ui\org.eclipse.smarthome.ui.paper\target\npm_cache\_git-remotes\_templates --mirror https://github.com/jquery/jquery-ui.git C:\eclipse\eclipse-smarthome2\git\smarthome\extensions\ui\org.eclipse.smarthome.ui.paper\target\npm_cache\_git-remotes\git-https-github-com-jquery-jquery-ui-git-1-11-2-5b6c4032fd076368d4b39dec471a4d07: Cloning into bare repository 'C:\eclipse\eclipse-smarthome2\git\smarthome\extensions\ui\org.eclipse.smarthome.ui.paper\target\npm_cache\_git-remotes\git-https-github-com-jquery-jquery-ui-git-1-11-2-5b6c4032fd076368d4b39dec471a4d07'...
[ERROR] npm ERR! git clone --template=C:\eclipse\eclipse-smarthome2\git\smarthome\extensions\ui\org.eclipse.smarthome.ui.paper\target\npm_cache\_git-remotes\_templates --mirror https://github.com/jquery/jquery-ui.git C:\eclipse\eclipse-smarthome2\git\smarthome\extensions\ui\org.eclipse.smarthome.ui.paper\target\npm_cache\_git-remotes\git-https-github-com-jquery-jquery-ui-git-1-11-2-5b6c4032fd076368d4b39dec471a4d07: fatal: bad object 22efbf5843bae3c5d84fc30cda328391c5fa4aec
[ERROR] npm ERR! git clone --template=C:\eclipse\eclipse-smarthome2\git\smarthome\extensions\ui\org.eclipse.smarthome.ui.paper\target\npm_cache\_git-remotes\_templates --mirror https://github.com/jquery/jquery-ui.git C:\eclipse\eclipse-smarthome2\git\smarthome\extensions\ui\org.eclipse.smarthome.ui.paper\target\npm_cache\_git-remotes\git-https-github-com-jquery-jquery-ui-git-1-11-2-5b6c4032fd076368d4b39dec471a4d07: fatal: remote did not send all necessary objects
[ERROR] npm ERR! Windows_NT 10.0.10586
[ERROR] npm ERR! argv "C:\\eclipse\\eclipse-smarthome2\\git\\smarthome\\extensions\\ui\\org.eclipse.smarthome.ui.paper\\node\\node.exe" "C:\\eclipse\\eclipse-smarthome2\\git\\smarthome\\extensions\\ui\\org.eclipse.smarthome.ui.paper\\node\\node_modules\\npm\\bin\\npm-cli.js" "install"
[ERROR] npm ERR! node v5.4.0
[ERROR] npm ERR! npm  v3.3.12
[ERROR] npm ERR! code 128
[ERROR] 
[ERROR] npm ERR! Command failed: git -c core.longpaths=true clone --template=C:\eclipse\eclipse-smarthome2\git\smarthome\extensions\ui\org.eclipse.smarthome.ui.paper\target\npm_cache\_git-remotes\_templates --mirror https://github.com/jquery/jquery-ui.git C:\eclipse\eclipse-smarthome2\git\smarthome\extensions\ui\org.eclipse.smarthome.ui.paper\target\npm_cache\_git-remotes\git-https-github-com-jquery-jquery-ui-git-1-11-2-5b6c4032fd076368d4b39dec471a4d07
[ERROR] npm ERR! Cloning into bare repository 'C:\eclipse\eclipse-smarthome2\git\smarthome\extensions\ui\org.eclipse.smarthome.ui.paper\target\npm_cache\_git-remotes\git-https-github-com-jquery-jquery-ui-git-1-11-2-5b6c4032fd076368d4b39dec471a4d07'...
[ERROR] npm ERR! fatal: bad object 22efbf5843bae3c5d84fc30cda328391c5fa4aec
[ERROR] npm ERR! fatal: remote did not send all necessary objects
[ERROR] npm ERR! 
[ERROR] npm ERR! 
[ERROR] npm ERR! If you need help, you may report this error at:
[ERROR] npm ERR!     <https://github.com/npm/npm/issues>
[ERROR] 
[ERROR] npm ERR! Please include the following file with any support request:
[ERROR] npm ERR!     C:\eclipse\eclipse-smarthome2\git\smarthome\extensions\ui\org.eclipse.smarthome.ui.paper\npm-debug.log
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 01:03 min
[INFO] Finished at: 2016-10-31T22:26:54+01:00
[INFO] Final Memory: 60M/834M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal com.github.eirslett:frontend-maven-plugin:1.0:npm (npm install) on project org.eclipse.smarthome.ui.paper: Failed to run task: 'npm install' failed. (error code 1) -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
````
I use Windows 10 and

    git --version
    git version 2.10.1.windows.1

and

    mvn --version
    Apache Maven 3.3.9 (bb52d8502b132ec0a5a3f4c09453c07478323dc5; 2015-11-10T17:41:47+01:00)
    Maven home: C:\Program Files (x86)\Maven\apache-maven-3.3.9\bin\..
    Java version: 1.8.0_101, vendor: Oracle Corporation
    Java home: C:\Program Files\Java\jdk1.8.0_101\jre
    Default locale: de_DE, platform encoding: Cp1252
    OS name: "windows 10", version: "10.0", arch: "amd64", family: "dos"


If someone could give me a hint how to build this, I would happily add some more details to the IDE setup guide.

--- a/docs/documentation/development/ide.md
+++ b/docs/documentation/development/ide.md
@@ -33,6 +33,7 @@ The Eclipse IDE is used for Eclipse SmartHome developments. The Eclipse Installe
 ![Step 7](images/ide7.png)
 10. Your workspace should now fully compile and you can start the runtime by launching the "SmartHome Runtime" launch configuration:
 ![Step 8](images/ide8.png)
+11. Access the PaperUI at [http://localhost:8080/paperui/index.html](http://localhost:8080/paperui/index.html). For more information about PaperUI see [PaperIU Development](notes.html#paperui-development-jshtml). 
 
 Note that you will find the sources in a subfolder called "git" within your selected installation folder. You can use any kind of git client here, if you do not want to use the git support from within the Eclipse IDE.
 If you want to push changes, you need to do so to [your personal fork of the Eclipse SmartHome repository](https://github.com/eclipse/smarthome/fork) in order to create a pull request. You will find more details in the ["How to contribute"](../community/contributing.html) documentation.

--- a/extensions/ui/org.eclipse.smarthome.ui.paper/PaperUI Setup.launch
+++ b/extensions/ui/org.eclipse.smarthome.ui.paper/PaperUI Setup.launch
@@ -1,0 +1,18 @@
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
+<launchConfiguration type="org.eclipse.m2e.Maven2LaunchConfigurationType">
+<booleanAttribute key="M2_DEBUG_OUTPUT" value="false"/>
+<stringAttribute key="M2_GOALS" value="clean install"/>
+<booleanAttribute key="M2_NON_RECURSIVE" value="false"/>
+<booleanAttribute key="M2_OFFLINE" value="false"/>
+<stringAttribute key="M2_PROFILES" value=""/>
+<listAttribute key="M2_PROPERTIES"/>
+<stringAttribute key="M2_RUNTIME" value="EMBEDDED"/>
+<booleanAttribute key="M2_SKIP_TESTS" value="false"/>
+<intAttribute key="M2_THREADS" value="1"/>
+<booleanAttribute key="M2_UPDATE_SNAPSHOTS" value="false"/>
+<stringAttribute key="M2_USER_SETTINGS" value=""/>
+<booleanAttribute key="M2_WORKSPACE_RESOLUTION" value="false"/>
+<booleanAttribute key="org.eclipse.jdt.launching.ATTR_USE_START_ON_FIRST_THREAD" value="true"/>
+<stringAttribute key="org.eclipse.jdt.launching.JRE_CONTAINER" value="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/JRE for JavaSE-1.8"/>
+<stringAttribute key="org.eclipse.jdt.launching.WORKING_DIRECTORY" value="${workspace_loc:/org.eclipse.smarthome.ui.paper}"/>
+</launchConfiguration>

--- a/targetplatform/EclipseSmartHome.setup
+++ b/targetplatform/EclipseSmartHome.setup
@@ -2413,6 +2413,10 @@
         content="&lt;?xml version=&quot;1.0&quot; encoding=&quot;UTF-8&quot;?>&#xA;&lt;lifecycleMappingMetadata>&#xA;  &lt;pluginExecutions>&#xA;    &lt;pluginExecution>&#xA;      &lt;pluginExecutionFilter>&#xA;        &lt;groupId>org.codehaus.mojo&lt;/groupId>&#xA;        &lt;artifactId>build-helper-maven-plugin&lt;/artifactId>&#xA;        &lt;versionRange>1.8&lt;/versionRange>&#xA;        &lt;goals>&#xA;          &lt;goal>add-source&lt;/goal>&#xA;          &lt;goal>add-test-source&lt;/goal>&#xA;        &lt;/goals>&#xA;      &lt;/pluginExecutionFilter>&#xA;      &lt;action>&#xA;        &lt;ignore />&#xA;      &lt;/action>&#xA;    &lt;/pluginExecution>&#xA;  &lt;/pluginExecutions>&#xA;&lt;/lifecycleMappingMetadata>&#xA;"
         targetURL="${workspace.location|uri}/.metadata/.plugins/org.eclipse.m2e.core/lifecycle-mapping-metadata.xml"
         encoding="UTF-8"/>
+    <setupTask
+        xsi:type="launching:LaunchTask"
+        excludedTriggers="STARTUP"
+        launcher="PaperUI Setup"/>
     <stream
         name="master"/>
   </project>

